internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -95,8 +100,10 @@ func proxyURLForTarget(address string) (*url.URL, error) {
 //     resolution is enabled using the dial option.
 func New(target resolver.Target, cc resolver.ClientConn, opts resolver.BuildOptions, targetResolverBuilder resolver.Builder, targetResolutionEnabled bool) (resolver.Resolver, error) {
 	r := &delegatingResolver{
-		target: target,
-		cc:     cc,
+		target:               target,
+		cc:                   cc,
+		proxyResolverCreated: false,
+		resolveNowCalled:     false,
```

Comment:
Nit: fields initialized to their zero values can be omitted, unless you feel that improves the readability of the code.